sound/soc/sof/sof-acpi-dev.c

```diff
@@ -191,6 +213,16 @@ static int sof_acpi_probe(struct platform_device *pdev)
 
 	dev_dbg(&pdev->dev, "ACPI DSP detected");
 
+	/*
+	 * Minnowboard has a known issue with the BIOS that prevents the DSP
+	 * from resuming back to D0 when the system resumes from S3 to S0.
+	 * So, detach the platform device from the pm_domain to prevent
+	 * the DSP from entering D3.
+	 */
+	if (dmi_check_system(byt_quirk_table))
+		if (dev->pm_domain)
+			dev_pm_domain_detach(dev, false);
```

Comment:
maybe add a comment that the second argument is telling the PM core to leave the DSP on during the detach process, not that it invalidates the detach itself:
@power_off: Used to indicate whether we should power off the device.